components/tdisplays3/display.py

```diff
@@ -28,10 +28,45 @@ def validate_tdisplays3(config):
 )
 
 async def to_code(config):
+    # Add platformio build_flags for the correct TFT_eSPI settings for the T-Display-S3
+    # This allows using current, unpatched versions of TFT_eSPI
+    cg.add_build_flag("-DUSER_SETUP_LOADED=1")
+    cg.add_build_flag("-DST7789_DRIVER=1")
+    cg.add_build_flag("-DINIT_SEQUENCE_3=1")
+    cg.add_build_flag("-DCGRAM_OFFSET")
+    cg.add_build_flag("-DTFT_RGB_ORDER=TFT_RGB")
+    cg.add_build_flag("-DTFT_INVERSION_ON=1")
+    cg.add_build_flag("-DTFT_PARALLEL_8_BIT=1")
+    cg.add_build_flag("-DTFT_WIDTH=170")
+    cg.add_build_flag("-DTFT_HEIGHT=320")
+    cg.add_build_flag("-DTFT_DC=7")
+    cg.add_build_flag("-DTFT_RST=5")
+    cg.add_build_flag("-DTFT_WR=8")
+    cg.add_build_flag("-DTFT_RD=9")
+    cg.add_build_flag("-DTFT_D0=39")
+    cg.add_build_flag("-DTFT_D1=40")
+    cg.add_build_flag("-DTFT_D2=41")
+    cg.add_build_flag("-DTFT_D3=42")
+    cg.add_build_flag("-DTFT_D4=45")
+    cg.add_build_flag("-DTFT_D5=46")
+    cg.add_build_flag("-DTFT_D6=47")
+    cg.add_build_flag("-DTFT_D7=48")
+    cg.add_build_flag("-DLOAD_GLCD=1")
```

Comment:
This is probably also related to fonts (see next comment); you might want to try whether it is really needed.
```suggestion
    cg.add_build_flag("-DLOAD_GLCD=1")
```

Reply:
Hi @guillempages - I'm still puzzled that yours doesn't display anything.  Thanks for the suggestions on define cleanup.  On the font defines - I may be using them in my next version, but can look into pulling them out of this version.  I'll continue to play around with this.  I guess it's possible there's something different between the touch and non-touch versions - however the first version I came up with worked for you.  There's not much different here except this version doesn't expose the TFT and SPR objects and the defines are now handled in the Python file.  

I might come up with a special branch for you to try that would just enable some debug logging around the display driver that is being loaded.  But let's definitely hold on merging this PR since it's still not working for @guillempages.